java-checks/src/main/java/org/sonar/java/checks/spring/TransactionalMethodVisibilityCheck.java

```diff
@@ -36,6 +40,8 @@ public class TransactionalMethodVisibilityCheck extends IssuableSubscriptionVisi
     "org.springframework.transaction.annotation.Transactional", "@Transactional",
     "org.springframework.scheduling.annotation.Async", "@Async");
 
+  private boolean isSpring6OrLater = false;
```

Comment:
From a few issues with non-primitve attributes in the past, and a bit of superstition ever since, we try to avoid having state in the check. Could we imagine removing the field, replacing it with a local variable in `visitNode` and passing that local variable to a static version of `requiredVisibilityMessage`?

Reply:
The field is initialized in `isCompatibleWithDependencies` so it can't be made local. To be able to get rid of that state we would need to change how the `DependencyVersionAware` interface works and use the `JavaFileScannerContext` to store that information.